processing/src/main/java/org/apache/druid/query/DataSource.java

```diff
@@ -118,6 +123,27 @@ public interface DataSource
    */
   DataSource withUpdatedDataSource(DataSource newSource);
 
+  default DataSource mapWithRestriction(Map<String, Optional<DimFilter>> rowFilters)
+  {
+    return mapWithRestriction(rowFilters, true);
+  }
+
+  /**
+   * Returns an updated datasource based on the policy restrictions on tables. If this datasource contains no table, no
+   * changes should occur.
+   *
+   * @param rowFilters a mapping of table names to row filters, every table in the datasource tree must have an entry
```

Comment:
Takes Map<String, Optional<Policy>> now, since policy is optional for druid tables, ppl can config any (or all) tables to be policy restricted. It's a single policy, since policy is returned from authorizer and it's a merged result of (policy rule, or policy template, which could be a serialized format and supports crud and stuff).